cloudflare-pages/action.yml

```diff
@@ -0,0 +1,62 @@
+name: 'Meticulous - Trigger tests for Cloudflare Pages'
+description: 'Trigger Meticulous tests for a Cloudflare Pages deployment'
+inputs:
+  github-token:
+    description: 'A GitHub Actions token (should have write permission on deployments)'
+    required: true
+  cloudflare-api-token:
+    description: 'API token for Cloudflare (should have Read permission for Cloudflare Pages)'
+    required: true
+  cloudflare-account-id:
+    description: 'Account ID for Cloudflare Pages (from the Workers & Pages section of the dashboard)'
+    required: true
+  cloudflare-project-name:
+    description: 'Name of your Cloudflare Pages project (from the Workers & Pages section of the dashboard)'
+    required: true
+  sleep-seconds:
+    description: 'Time to sleep before starting to poll deployment status (in seconds)'
+    required: false
+    default: "300"
+  deployment-poll-seconds:
+    description: 'Time to wait between API calls to poll for the deployment to be ready (in seconds)'
+    required: false
+    default: "60"
+  wait-till-ready-seconds:
+    description: 'Maximum time to wait for the deployment to be ready (in seconds)'
+    required: false
+    default: "3600"
+runs:
+  using: "composite"
+  steps:
+    - name: Sleep a bit to wait for the deployment to exist
+      if: ${{ inputs.sleep-time != '0' }}
+      shell: bash
+      run: sleep ${{ inputs.sleep-time }}
+    - name: Wait for deployment to be ready and get URL
+      shell: bash
+      id: get-url
+      run: |
+        STATUS="unknown"
+        SLEPT=0
+        while [[ "$STATUS" != "success" ]]; do
+          echo "Checking deployment status..."
+          STATUS=$(curl -s -X GET "https://api.cloudflare.com/client/v4/accounts/${{ inputs.cloudflare-account-id }}/pages/projects/${{ inputs.cloudflare-project-name }}/deployments" -H "Authorization: Bearer ${{ inputs.cloudflare-api-token }}" -H "Content-Type:application/json" | jq -c 'first(.result[] | select(.deployment_trigger.metadata.commit_hash == "${{ github.sha }}") | .latest_stage.status)' --raw-output | tr -d '\n')
+          if [[ "$STATUS" != "success" ]]; then
+            if [[ $SLEPT -gt ${{ inputs.wait-till-ready-seconds }} ]]; then
+              echo "Deployment is still not ready, failing..."
+              exit 1
+            fi
+            echo "Deployment is not ready yet, sleeping..."
+            sleep ${{ inputs.deployment-poll-seconds }}
+            SLEPT=$((SLEPT+${{ inputs.deployment-poll-seconds }}))
+          fi
+        done
+        echo "Deployment is ready!"
+        echo "url=`curl -s -X GET "https://api.cloudflare.com/client/v4/accounts/${{ inputs.cloudflare-account-id }}/pages/projects/${{ inputs.cloudflare-project-name }}/deployments" -H "Authorization: Bearer ${{ inputs.cloudflare-api-token }}" -H "Content-Type:application/json" | jq -c 'first(.result[] | select(.deployment_trigger.metadata.commit_hash == "${{ github.sha }}") | .url)' --raw-output | tr -d '\n'`" >> $GITHUB_OUTPUT
```

Comment:
does `cloudflare-project-name` (or any other part) need to be URL encoded?

Reply:
No, the project name is alphanumeric plus `-` which doesn't need to be encoded. The others are all just alphanumeric.